src/ui/MessageInput/hooks/usePaste/index.ts

```diff
@@ -20,8 +20,15 @@ export function usePaste({
   setHeight,
   channel,
   setMentionedUsers,
+  customOnPaste,
+  onChange
 }: DynamicProps): (e: React.ClipboardEvent<HTMLDivElement>) => void {
   return useCallback((e) => {
+    if (customOnPaste && customOnPaste(e)) {
+      // custom logic already handled input, exit here
+      return;
+    }
```

Comment:
is customOnPaste conditionally passed in?

Reply:
`customOnPaste` from our flows is always defined. This entire function `usePaste` is hooked into the `onPaste` of the input `div` component, so our callback gets called every time there's a paste event in the `div`. The idea here is that we have a nice way of getting the callback first and handling it ourselves before letting UIKit do its thing.